catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DavEntry.java

```diff
@@ -0,0 +1,317 @@
+            FileUtils.deleteQuietly(file);
+        }
+
+        // Return if there are changes
+        return isExists() != origExists || getLastModified() != origLastModified
+                || isDirectory() != origDirectory || getLength() != origLength || !Objects.equals(
+                geteTag(),
+                origEtag);
+    }
+
+    DavEntry newChildInstance(String location) {
+        return new DavEntry(this, location);
+    }
+
+    // construct the fully qualified location from a fully qualified parent and
+    // a child relative to the parent
+    static String getLocation(String initialLocation, DavEntry parent) {
+        String location = initialLocation;
+        if (parent != null && !location.startsWith("http")) {
+            String parentLocation = parent.getLocation();
+            if (parentLocation.endsWith("/") && location.startsWith("/")) {
+                location = location.replaceFirst("/", "");
+            }
+            if (!parentLocation.endsWith("/") && !location.startsWith("/")) {
+                location = "/" + location;
+            }
+            location = parentLocation + location;
+        }
+        try {
+            URL url = new URL(location);
+            URI uri = new URI(url.getProtocol(),
+                    url.getUserInfo(),
+                    url.getHost(),
+                    url.getPort(),
+                    url.getPath(),
+                    url.getQuery(),
+                    url.getRef());
+            location = uri.toASCIIString();
+        } catch (MalformedURLException | URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+        return location;
+    }
+
+    /**
+     * Return the parent entry.
+     *
+     * @return the parent entry
+     */
+    public DavEntry getParent() {
+        return parent;
+    }
+
+    /**
+     * Return the level
+     *
+     * @return the level
+     */
+    int getLevel() {
+        return getParent() == null ? 0 : getParent().getLevel() + 1;
+    }
+
+    /**
+     * Return the directory's files.
+     *
+     * @return This directory's files or an empty
+     * array if the file is not a directory or the
+     * directory is empty
+     */
+    DavEntry[] getChildren() {
+        return children != null ? children : getEmptyEntries();
+    }
+
+    /**
+     * Set the directory's files.
+     *
+     * @param children This directory's files, may be null
+     */
+    void setChildren(final DavEntry... children) {
+        this.children = children;
+    }
+
+    /**
+     * Return a local cache of the file being monitored.
+     * This file is invalidated if necessary when refresh() is called.
+     *
+     * @param sardine
+     * @return the file being monitored
+     */
+    public File getFile(Sardine sardine) throws IOException {
+        if (file == null || !file.exists()) {
+            Path dav = Files.createTempDirectory("dav");
```

Comment:
Added an on-completion hook to remove the temporary cache after it is successfully processed.